src/vue/pages/TradeExchange.vue

```diff
@@ -0,0 +1,255 @@
+            :page-limit="recordsToShow"
+            @first-page-load="setTradeHistory"
+            @next-page-load="extendTradeHistory"
+          />
+        </div>
+      </div>
+
+      <div class="trade-exchange__orders">
+        <h2 class="app__table-title">
+          {{ 'trade.orders-section-title' | globalize }}
+        </h2>
+        <div v-if="assets.base" class="trade-exchange__orders-wrapper">
+          <trade-orders
+            class="trade-exchange__orders-list"
+            :assets="assets"
+            :is-buy="true"
+            :is-loading="isBuyOrdersLoading"
+            :orders-list="buyOrdersList"
+            @reload-trades="componentConfig.isNeededToReloadData = true"
+          />
+
+          <trade-orders
+            class="trade-exchange__orders-list"
+            :assets="assets"
+            :is-buy="false"
+            :is-loading="isSellOrdersLoading"
+            :orders-list="sellOrdersList"
+            @reload-trades="componentConfig.isNeededToReloadData = true"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Chart from '@/vue/common/chart/Chart'
+import TradeHistory from '@/vue/pages/TradeExchange/Trade.History'
+import TradeOrders from '@/vue/pages/TradeExchange/Trade.Orders'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import config from '@/config'
+import { Sdk } from '@/sdk'
+import { SECONDARY_MARKET_ORDER_BOOK_ID } from '@/js/const/offers'
+import CollectionLoader from '@/vue/common/CollectionLoader'
+
+export default {
+  name: 'trade-exchange',
+  components: {
+    Chart,
+    TradeHistory,
+    TradeOrders,
+    CollectionLoader,
+  },
+  props: {
+    // prop from parent router-view component
+    componentConfig: { type: Object, required: true },
+  },
+  data: () => ({
+    assets: {
+      base: '',
+      quote: '',
+    },
+    tradeHistory: [],
+    buyOrdersList: [],
+    sellOrdersList: [],
+    isTradeHistoryLoading: false,
+    isBuyOrdersLoading: false,
+    isSellOrdersLoading: false,
+    recordsOrder: 'desc',
+    recordsToShow: 10,
+    config,
+  }),
+  watch: {
+    async 'componentConfig.isNeededToReloadData' (status) {
+      if (status) {
+        await this.loadTradeOrders()
+        await this.loadTradeHistory()
+        // says to Trade component that current loaded data is actual
+        this.componentConfig.isNeededToReloadData = false
+      }
+    },
+    'componentConfig.assets': {
+      deep: true,
+      handler: function (assets) {
+        this.setCurrentAssets(assets)
+        if (assets.base && assets.quote) {
+          this.loadTradeOrders()
+          this.loadTradeHistory()
+        }
+      },
+    },
+  },
+  created () {
+    this.setCurrentAssets(this.componentConfig.assets)
+    if (this.componentConfig.assets.base) {
+      this.loadTradeOrders()
+      this.loadTradeHistory()
+    }
+  },
+  methods: {
+    async loadTradeHistory () {
```

Comment:
move it to trade-history component

Reply:
Correct me if I'm wrong: if we have component, that requires some data, that can be reloaded by an event from another component – we need to move data loading to the parent component. In other cases, we load data inside component?

Reply:
just call the method that fetches the new data instead of toggling of isTradeHistoryLoading

Reply:
just rename it to renderer